spec/controllers/flexible_criteria_controller_spec.rb

```diff
@@ -0,0 +1,566 @@
+        end
+      end
+
+      context 'without error on an assignment that already has criteria' do
+        before(:each) do
+          post_as @admin, :create,
+                  format: :js,
+                  assignment_id: @assignment.id,
+                  flexible_criterion: {flexible_criterion_name: 'first',
+                                       max: 10}
+        end
+        it 'should respond with appropriate content' do
+          expect(assigns(:criterion)).to be_truthy
+          expect(assigns(:assignment)).to be_truthy
+        end
+        it 'should render the create_and_edit template' do
+          is_expected.to render_template(:'flexible_criteria/create_and_edit')
+        end
+
+        it 'should respond with success' do
+          is_expected.to respond_with(:success)
+        end
+      end
+    end
+
+    context '#download' do
+      before(:each) do
+        post_as @admin, :download, :assignment_id => @assignment.id
+      end
+
+      it 'should respond with success' do
+        is_expected.to respond_with(:success)
+      end
+
+      it 'should respond with appropriate content' do
+        expect(response.header['Content-Type']).to eql('text/csv')
+        expect(@response.body).to eql(FLEXIBLE_CRITERIA_CSV_STRING)
+        expect(assigns(:assignment)).to be_truthy
+      end
+    end
+
+    context '#upload' do
+      context 'with file containing incomplete records' do
+        before(:each) do
+          tempfile = fixture_file_upload('/files/flexible_incomplete.csv')
+          post_as @admin, :upload, :assignment_id => @assignment.id,
+                  :upload => {:flexible => tempfile}
+        end
+        it 'should respond with appropriate content' do
+          expect(assigns(:assignment)).to be_truthy
+        end
+
+        it 'should set the flash' do
+          is_expected.to set_the_flash
+        end
+
+        it 'should respond with redirect' do
+          is_expected.to respond_with(:redirect)
+        end
+
+      end
+
+      context 'with file containing partial records' do
+        before(:each) do
+          tempfile = fixture_file_upload('/files/flexible_partial.csv')
+          post_as @admin,
+                  :upload,
+                  :assignment_id => @assignment.id,
+                  :upload => {:flexible => tempfile}
+        end
+        it 'should respond with appropriate content' do
+          expect(assigns(:assignment)).to be_truthy
+        end
+        it 'should set the flash' do
+          is_expected.to set_the_flash
+        end
+
+        it 'should respond with redirect' do
+          is_expected.to respond_with(:redirect)
+        end
+      end
+
+      context 'with file containing full records' do
+        before(:each) do
+          FlexibleCriterion.destroy_all
+          tempfile = fixture_file_upload('/files/flexible_upload.csv')
+          post_as @admin, :upload, :assignment_id => @assignment.id,
+                  :upload => {:flexible => tempfile}
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.